frappe/email/receive.py

```diff
@@ -405,10 +413,12 @@ def set_subject(self):
 		"""Parse and decode `Subject` header."""
 		_subject = decode_header(self.mail.get("Subject", "No Subject"))
 		self.subject = _subject[0][0] or ""
+		charset = _subject[0][1]
 
-		if _subject[0][1]:
+		if charset:
 			# Encoding is known by decode_header (might also be unknown-8bit)
-			self.subject = safe_decode(self.subject, _subject[0][1])
+			charset = email.charset.ALIASES.get(charset, charset)
+			self.subject = safe_decode(self.subject, charset)
```

Comment:
Did this call to `safe_decode` cause a LookupError as well?

Was thinking that instead of updating the aliases dictionary, we could have a separate dictionary with these "missed" charsets, and reference that when we face a LookupError here

Reply:
This is safe_decode. No, it does not return LookupError as it will do nothing when exception. Can you explain a bit more what you mean?
```
def safe_decode(param, encoding="utf-8"):
	try:
		param = param.decode(encoding)
	except Exception:
		pass
	return param
```

Reply:
Hey, pushed a commit explaining what I meant. Let me know if this approach works!

Reply:
> Actually, I was thinking that the `safe_decode` can do a bit more with LookupError. But since it was deep into the core, so I don't touch it yet. Or should it be something like this?
> 
> ```
> def safe_decode(param, encoding="utf-8"):
> 	try:
> 		param = param.decode(encoding)
>         except LookupError:
>                 # Try again, if lookup error
>                 encoding = ALIASES.get(encoding, encoding)
>                 param = param.decode(encoding)
> 	except Exception:
> 		pass
> 	return param
> ```

IMHO, the `safe_decode` should be fixed to handled this, and all decode use it to be "safe".

Reply:
> Hey, pushed a commit explaining what I meant. Let me know if this approach works!

OH I just saw our commit. Yes, it should works too. I am ok with this if you want to go ahead.
But, do you think my comment on safe_decode is valid? Or we shouldn't touch the safe_decode?

Reply:
Could add an optional parameter to pass our alternate mapping to safe_decode to be used in case of LookupErrors, yeah. I'll push the changes and let you know, once you confirm it's all good will merge.

Reply:
@kittiu please test the current state of the PR, if all good will merge it.